devito/interfaces.py

```diff
@@ -429,7 +452,20 @@ def _allocate_memory(self):
         self._full_data = self._data.view()
 
         if self.pad_time:
-            self._data = self._data[self.time_order:, :, :]
+            self._data = self._data[self.time_order:, ...]
+
+    def indexify(self):
+        """Convert base symbol and dimensions to indexed data accesses
+
```

Comment:
Agreed. However, Michael is pointing out that this is not the right fix. So maybe we should think about how to properly do this.